pkg/store/kotsstore/downstream_store.go

```diff
@@ -508,6 +523,59 @@ func getDownstreamVersionStatus(status types.DownstreamVersionStatus, hasError s
 	return s
 }
 
+func needsKotsUpgrade(app *kotsv1beta1.Application) bool {
+	if app == nil {
+		return false
+	}
+	if app.Spec.MinKotsVersion == "" && app.Spec.TargetKotsVersion == "" {
+		return false
+	}
+
+	featureEnabled := false
+	for _, f := range app.Spec.ConsoleFeatureFlags {
+		if f == "admin-console-auto-updates" {
+			featureEnabled = true
+			break
+		}
+	}
+
+	if !featureEnabled {
+		return false
+	}
+
+	var minKotsSemver, targetKotsSemver, currentKotsSemver *semver.Version
```

Comment:
why not just use the `IsKotsVersionCompatibleWithApp` function that already eixsts?